src/components/common/UserProfileEdit/UserProfileEdit.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { Tab, TabList, TabPanel, Tabs } from '~shared/Tabs';
+
+import ProfileTemplate from '~frame/ProfileTemplate';
+import LandingPage from '~frame/LandingPage';
+
+import { useAppContext } from '~hooks';
+
+import UserProfileSpinner from '../UserProfile/UserProfileSpinner';
+import Sidebar from './Sidebar';
+import UserMainSettings from './UserMainSettings';
+import UserAdvancedSettings from './UserAdvancedSettings';
+
+const displayName = 'common.UserProfileEdit';
+
+const MSG = defineMessages({
+  headingMain: {
+    id: `${displayName}.headingMain`,
+    defaultMessage: 'User profile',
+  },
+  headingAdvance: {
+    id: `${displayName}.headingAdvance`,
+    defaultMessage: 'Advanced settings',
+  },
+});
+
+const UserProfileEdit = () => {
+  const { user, userLoading } = useAppContext();
+
+  if (userLoading || user === null) {
```

Comment:
After rebasing,  `null` and `undefined` should be swapped so it loads when `undefined` and redirects if it returns without a user (`null`)